ui-admin/src/util/tableUtils.tsx

```diff
@@ -1,23 +1,101 @@
 import React, { HTMLProps, useState } from 'react'
-import { flexRender, Header, Table } from '@tanstack/react-table'
+import { Column, flexRender, Header, Table } from '@tanstack/react-table'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import { faCaretDown, faCaretUp, faColumns } from '@fortawesome/free-solid-svg-icons'
 
+/**
+ * Returns a debounced input react component
+ * Adapted from https://tanstack.com/table/v8/docs/examples/react/filters
+ */
+function DebouncedInput({
+  value: initialValue,
+  onChange,
+  debounce = 200,
+  ...props
+}: {
+  value: string
+  onChange: (value: string) => void
+  debounce?: number
+} & Omit<React.InputHTMLAttributes<HTMLInputElement>, 'onChange'>) {
+  const [value, setValue] = React.useState(initialValue)
+
+  React.useEffect(() => {
```

Comment:
Style nit: generally, the convention is to import these instead of using them from the React namespace.

Reply:
Resolved in 7bfb73d61863d529779ed3379e83b923eef5c498